packages/platform/types/bookings/2024-08-13/inputs/create-booking-input.pipe.ts

```diff
@@ -0,0 +1,111 @@
+import type { PipeTransform } from "@nestjs/common";
+import { Injectable, BadRequestException } from "@nestjs/common";
+import { plainToClass } from "class-transformer";
+import type { ValidationError } from "class-validator";
+import { validateSync } from "class-validator";
+
+import { CreateRecurringBookingInput_2024_08_13 } from "./create-booking.input";
+import { CreateBookingInput_2024_08_13 } from "./create-booking.input";
+import { CreateInstantBookingInput_2024_08_13 } from "./create-booking.input";
+
+export type CreateBookingInput =
+  | CreateBookingInput_2024_08_13
+  | CreateRecurringBookingInput_2024_08_13
+  | CreateInstantBookingInput_2024_08_13;
```

Comment:
this discriminative union can be improved, the instant field can be used as the discriminative field

Reply:
I think it's ok to leave it as is - there are no compilation problems and I think it's cleaner to check `if ("instant" in body` than `CreateBookingInput_2024_08_13` and `CreateRecurringBookingInput_2024_08_13` having property `instant` that they do not rely on nor need to know about just to validate which input it is when it is working already, so will leave as is unless there is a problem / error.

Reply:
I agree that in the current state it's fine to leave as id. I believe discriminative union is almost always  the right approach, giving more context to typescript ensure that developers in the future maintaining the code or adding new features don't encounter typing errors. creating union of types without any discriminative factor is prone to triggering TS errors that are hard to solve for junior engineers

Reply:
and props that are not required and just used to give context can be excluded from doc